V2.0.0 - MPhys Variable Naming Convention Classes (#181)
OpenMDAO/mphys: docs/basics/model_hierarchy.rst

===============
Model Hierarchy
===============

MPhys uses a hierarchical set of OpenMDAO groups to build multiphysics models.
Each level of the hierarchy is a different OpenMDAO group that MPhys provides.

The highest level of the model is the multipoint group.

> The multipoint group consist of scenarios which represent different conditions and/or types of multiphysics analyses to performed.

Reviewer: ```suggestion
The multipoint group consists of scenarios which represent different conditions and/or types of multiphysics analyses to be performed.
```

Within the scenario is the coupling group which represents the primary multiphysics problem for the scenario.

> See :ref:`model_assembly` for information an out to create this model heirarchy for a particular problem.

Reviewer: ```suggestion
See :ref:`model_assembly` for guidance on how to create this model hierarchy for a particular problem.
```

The scenario group contains a coupling group and any scenario-specific computation that needs to occur before or after the associated coupled problem is solved.
The subsystems before the coupled problem are referred to as 'pre_coupling' subsystems.
The subsystems after the coupled problem are referred to as 'post_coupling' subsystems.
An example post_coupling subsystem would be a sonic boom propagation analysis after an aeropropulsive coupling analysis.

.. figure:: scenario_example.png
	:scale: 50 %
	:alt: Example Scenario

MPhys provides a library of these Scenario groups designed for specific type problems.
See :ref:`scenario_library` for details about specific standardized scenarios.
If a particular multiphysics problem is not covered by the MPhys library, new scenarios and coupling groups can be created by subclassing the :class:`~mphys.MPhysGroup`.

-----------------
Multipoint Groups
-----------------

There are two versions of the multipoint group:

1. ``Multipoint`` is derived from the standard OpenMDAO ``Group``
2. ``MultipointParallel`` is derived for the OpenMDAO ``ParallelGroup``.

> For both versions have a function,  :func:`~mphys.Multipoint.mphys_add_scenario`, is used to populate

Reviewer: ```suggestion
For both versions, a function :func:`~mphys.Multipoint.mphys_add_scenario` is used to populate
```
